### PR TITLE
update documentation line indicating all List items must be same size

### PR DESCRIPTION
**Pull request by hkparker**
### Description:

Remove the line in the documentation that indicates that `widget.List` can only contain items of the same size.  This is no longer true since the introduction of `SetItemHeight` in 2.3

Addresses the confusion that led to #4777.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Review by andydotxyz — CHANGES_REQUESTED** [review]
I would not remove it, though I agree it is no longer correct.
How about indicating that /by default/ all items are the same size, but that `SetItemHeight` can be used to change this for a specific row?

**Comment by hkparker** [comment]
yeah good idea, let me know how this language looks

**Comment by Jacalz** [comment]
Please don't force-push. It usually breaks GitHub's ability to track comments from previous reviews., We'll squash on merge if we think it is needed.

Also FYI: It is a good idea to press the button to re-request a review after making changes :)

**Comment by hkparker** [comment]
Ach, sorry, I remembered squashing existed right after I did that.  It's been a while.  Let me know if you want me to open a PR without that.  I haven't checked the checklist because it hasn't happened yet (I didn't include tests for this of course, an the pipeline hasn't run yet so technically I don't know if the linters and tests passed).  But if I should just check everything since it's a comment change will do.  And thanks for the FYI on re-review :)

**Comment by Jacalz** [comment]
> Ach, sorry, I remembered squashing existed right after I did that. It's been a while. Let me know if you want me to open a PR without that.

Nah, it's fine. You had no open inline comments so nothing broke really.

> I haven't checked the checklist because it hasn't happened yet (I didn't include tests for this of course, an the pipeline hasn't run yet so technically I don't know if the linters and tests passed).

Your PR is obviously just updating a comment so this is more a theoretical problem than anything. However, a good practice is to run the tests and linting locally to test things out before opening a PR (at least if you are modifying code). The idea is to fill out the checkboxes before opening the PR. Tests and linting ought to still complete ;)

> And thanks for the FYI on re-review :)

No problem. We've all been there. Happy to help out :)

**Comment by hkparker** [comment]
Ah yeah those comments, ok cool well I'll avoid a force push in the future.

Got it, can run all the tests locally for any code change, thanks for the context.  Updated the PR description with checked boxes.

**Comment by Jacalz** [comment]
> Updated the PR description with checked boxes.

Now I'm really picky. Sorry about that, but you haven't added any tests so that checkbox should of course not be checked.

**Comment by hkparker** [comment]
Of course!  Updated the description

File: widget/list.go
Comment (Jacalz) [inline widget/list.go:26]: This is more grammatically correct:
```suggestion
// By default, List requires that all items are the same size, but specific
```

Comment (hkparker) [reply widget/list.go:26]: Ah, yeah that's better

**Review by Jacalz — APPROVED** [review]
Thanks. I can't merge until Andrew has re-reviewed but it looks great now. Thanks for the contribution and sorry for all a the back and forth :)

**Comment by dweymouth** [comment]
I dismissed Andy's review and went ahead and merged this since it's so straightforward

**Comment by Jacalz** [comment]
> I dismissed Andy's review and went ahead and merged this since it's so straightforward

Cool. Didn't know that was a thing 🙃

**Comment by hkparker** [comment]
@Jacalz Ah the back and forth is appreciated, good to learn what I should do next time!

**Comment by Jacalz** [comment]
Cool. Thanks for you contribution :)

**Pull request merged**
